src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -253,31 +254,42 @@ internal SymbolTreeInfo Create()
                     try
                     {
                         _metadataReader = moduleMetadata.GetMetadataReader();
+
+                        // First, walk all the symbols from metadata, populating the parentToChilren
+                        // map accordingly.
+                        GenerateMetadataNodes();
+
+                        // Now, once we populated the initial map, go and get all the inheritance 
+                        // information for all the types in the metadata.  This may refer to 
+                        // types that we haven't seen yet.  We'll add those types to the parentToChildren
+                        // map accordingly.
+                        PopulateInheritanceMap();
+
+                        // Clear the set of type definitions we read out of this piece of metadata.
+                        _allTypeDefinitions.Clear();
                     }
-                    catch (BadImageFormatException)
+                    catch (BadImageFormatException ex) when (ReportWatson(ex))
```

Comment:
Why not just directly call `FatalError.ReportWithoutCrash`?

Reply:
Also catch `IOException`?

Reply:
ah. yes. I can do that.